### DISCUSSION TITLE
Chromecast and GoogleDNS

### DISCUSSION THREAD

**Discussion by TMDWU** [DesktopECHO/Pi-hole-for-Android#29] [General]
I'm new here, and I'm completely impressed about this project, thank you so much for making this possible!

I'm using an old Android phone as Pi-hole **and** Hotspot at the same time and it works incredible. But I'm wondering if it's possible to block 8.8.8.8 and 8.8.4.4 from Raspbian directly and without modifying the router? I'm 100% sure my Chromecasts are using the Pi-hole but I still see ads, I guess it's bypassing something due to its hard-coded software nature.

"Why" would you ask? the idea is to have a portable router/repeater/ad-blocker that would work anywhere I go (for example, when visiting family) and I don't have any experience regarding IPtables.

No idea if this is already in the image or if it's a planned feature out of the box, thank you so much again.

**Comment by DesktopECHO** [comment]
Hi, glad to hear it's worked out for you.  I'm surprised this works actually, when your enable Hotspot on your phone Android will start up its own DNS/DHCP server and usually the hotspot fails because Pi-hole is already occupying that port.

I've never tried redirecting DNS at the gateway, but apparently you can do this with[ iptables](https://community.ui.com/questions/Intercepting-and-Re-Directing-DNS-Queries/cd0a248d-ca54-4d16-84c6-a5ade3dc3272).

**Reply by TMDWU** [reply]
I'm pretty impressed too, didn't expect it to work that well and it's still working to this day.

I've been playing with iptables (and fresh installations) all day and I have a weird issue, when I hit the "Stop" button on Pi Deploy, the phone automatically reboots when trying to stop the iptables (even if they worked and redirect the Chromecasts to the Pi-hole), I don't know why it happens but for now I'll enjoy the experience of having a phone/mobile repeater/ad-blocker all in one.

For additional information, I tried (another) 12 fresh installations without touching a thing and out of 2 of them, the Chromecasts used the Pi-Hole out of the box (don't know why it is so random) but ads were still playing even if it was filtering them, I suspect Google did some extra stuff these days to bypass ad-blockers.

Anyway, thanks for the link, I'll keep trying and return when I have some good news!


Edit regarding the Hotspot (for future references): for Pi-hole to start without any issues, you must disable Hotspot, turn on your Pi-hole then after that, start the hotspot.

**Reply by DesktopECHO** [reply]
> Edit regarding the Hotspot (for future references): for Pi-hole to start without any issues, you must disable Hotspot, turn on your Pi-hole then after that, start the hotspot.

...aha that makes sense.  The other thing that's allowing this to work now is a recent-ish change I made that allows [connectivitycheck.gstatic.com](https://connectivitycheck.gstatic.com/) to resolve before Unbound can reach the internet. 

Before that change, there was a chicken-and-egg scenario where Android would refuse to compete a WiFi join until connectivitycheck.gstatic.com would resolve, but Unbound couldn't resolve because there was no internet.
